mate-session/test-client-dbus.c

```diff
@@ -64,129 +58,138 @@ session_manager_connect (void)
                 }
         }
 
-        sm_proxy = dbus_g_proxy_new_for_name (bus_connection,
-                                              SM_DBUS_NAME,
-                                              SM_DBUS_PATH,
-                                              SM_DBUS_INTERFACE);
+        sm_proxy = g_dbus_proxy_new_sync (bus_connection,
+                                          G_DBUS_PROXY_FLAGS_NONE,
+                                          NULL,
+                                          SM_DBUS_NAME,
+                                          SM_DBUS_PATH,
+                                          SM_DBUS_INTERFACE,
+                                          NULL,
+                                          NULL);
         return (sm_proxy != NULL);
 }
 
 static void
-on_client_query_end_session (DBusGProxy     *proxy,
-                             guint           flags,
-                             gpointer        data)
+on_client_query_end_session (GDBusProxy     *proxy,
+                             GVariant   *parameters)
 {
         GError     *error;
         gboolean    is_ok;
-        gboolean UNUSED_VARIABLE res;
+        GVariant   *res;
+        guint       flags;
         const char *reason;
 
         is_ok = FALSE;
         reason = "Unsaved files";
+        g_variant_get (parameters, "(u)", &flags);
 
         g_debug ("Got query end session signal flags=%u", flags);
 
         error = NULL;
-        res = dbus_g_proxy_call (proxy,
-                                 "EndSessionResponse",
-                                 &error,
-                                 G_TYPE_BOOLEAN, is_ok,
-                                 G_TYPE_STRING, reason,
-                                 G_TYPE_INVALID,
-                                 G_TYPE_INVALID);
+        res = g_dbus_proxy_call_sync (proxy,
+                                      "EndSessionResponse",
+                                      g_variant_new ("(bs)",
+                                                     is_ok,
+                                                     reason),
+                                      G_DBUS_CALL_FLAGS_NONE,
+                                      -1,
+                                      NULL,
+                                      &error);
+        if (error != NULL) {
```

Comment:
Does `g_dbus_proxy_call_sync` set it to `NULL` on success, or just doesn't touch it in that case?
It's probably better to check the return value, like it's done in #170.

Reply:
Same in two similar situations below